### PR TITLE
Check for port before setting Origin header for watch requests.

### PR DESCRIPTION
**Pull request by kjcjohnson**
When the Kubernetes URL used doesn't specify a port
(e.g., https://example.com/api/v1/...), the origin header for
watch requests ends up with a port of -1 (e.g., https://example.com:-1).
This happens because calling `getPort()` on a java.net.URL object that
does not have a port explicitly specified will always return -1. The
return value was always just inserted into the origin header.

Now, we check for this and only append a port to the origin header if
`getPort()` returns something other than -1. We make this change in both
the WatchConnectionManager and WatchHTTPManager classes.

I wasn't able to come up with a way to make a test for this, because the 
mock Kubernetes server support specifying the port, and especially not 
the defaults of 80/443. 

Fixes issue #1667

### CODE REVIEW

**Comment by gastaldi** [comment]
Ok to test

**Comment by rohanKanojia** [comment]
ok to test

**Comment by cjellick** [comment]
Do we need an origin header at all? That’s really just useful for protecting browsers from cross site scripting.

Could we simply drop the header all together?

**Comment by rohanKanojia** [comment]
@kjcjohnson: Could you please test your patch without header? If it's not breaking anything then let's drop it.

**Comment by kjcjohnson** [comment]
It looks like dropping the origin header won't work. I took some Wireshark captures of the packets, and they're identical other than the Origin header, but only the request with the header succeeds:

Without Origin header:
```
GET /k8s/clusters/c-scffx/api/v1/namespaces/test-ns/pods?fieldSelector=metadata.name%3Dtestpod-1sntr&watch=true HTTP/1.1
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Key: 1Ass6sr+wPqYyEddMR7FPQ==
Sec-WebSocket-Version: 13
Authorization: Bearer <token>
Host: rancher.example.com
Accept-Encoding: gzip
User-Agent: okhttp/3.12.0

HTTP/1.1 403 Forbidden
X-Powered-By: ARR/3.0
X-Powered-By: ASP.NET
Date: Thu, 01 Aug 2019 19:00:43 GMT
Content-Length: 0
```

With Origin header:
```
GET /k8s/clusters/c-scffx/api/v1/namespaces/test-ns/pods?fieldSelector=metadata.name%3Dtestpod-rfw80&watch=true HTTP/1.1
Origin: http://rancher.example.com
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Key: zmA6ry8uoV5uumoYhh8rCg==
Sec-WebSocket-Version: 13
Authorization: Bearer <token>
Host: rancher.example.com
Accept-Encoding: gzip
User-Agent: okhttp/3.12.0

HTTP/1.1 101 Switching Protocols
Upgrade: websocket
Sec-WebSocket-Accept: WcnYoYGI9IQ0piiOUMVaDwgV1OU=
X-Powered-By: ARR/3.0
Connection: Upgrade
X-Powered-By: ASP.NET
Date: Thu, 01 Aug 2019 19:25:45 GMT
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ### CHANGELOG
 
 ### 4.3-SNAPSHOT
-  Bugs
+  Bugs
```
Comment (gastaldi) [inline CHANGELOG.md]:
```suggestion
  Bugs
```

**Comment by gastaldi** [comment]
[merge]

**Pull request merged**